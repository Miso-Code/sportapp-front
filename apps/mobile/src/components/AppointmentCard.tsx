import React from 'react'
import dayjs from 'dayjs'
import { useTranslation } from 'react-i18next'
import { View, StyleSheet } from 'react-native'
import { Card, Text, Button } from 'react-native-paper'

interface AppointmentCardProps {
	date: string
	trainer: string
	type: string
	location: string
}

const AppointmentCard: React.FC<AppointmentCardProps> = ({
	date,
	trainer,
	type,
	location,
	...props
}) => {
	const { t } = useTranslation()
	return (
		<Card elevation={1} style={styles.card} {...props}>
			<Card.Content style={styles.cardContent}>
				<Text variant='labelMedium'>
					{dayjs(date).format('DD/MM/YYYY hh:mm A')}
				</Text>
				<Text variant='titleMedium' style={styles.cardTitle}>
					{trainer}
				</Text>
				<Text variant='bodyMedium'>
					Tipo: {t(`preference.types.${type}`)}
				</Text>

				<View style={styles.chipContainer}>
					{type === 'virtual' ? (
						<Button mode='contained' onPress={() => {}} disabled>
							Ir a la cita
						</Button>
					) : (
						<Text testID='location' variant='bodyMedium'>
							Ubicaci√≥n: {location}
						</Text>
					)}
				</View>
			</Card.Content>
		</Card>
	)
}

const styles = StyleSheet.create({
	card: {
		backgroundColor: '#f2f2f2',
		width: '100%'
	},
	cardTitle: {
		fontWeight: 'bold'
	},
	cardContent: {
		justifyContent: 'center',
		marginTop: 10
	},
	chipContainer: {
		alignItems: 'flex-start',
		justifyContent: 'space-between',
		flexDirection: 'row',
		marginTop: 10
	}
})

export default AppointmentCard
