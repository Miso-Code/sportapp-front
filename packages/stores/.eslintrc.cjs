const eslintDefaultConfig = require('@sportapp/eslint-config/library')

module.exports = {
	...eslintDefaultConfig
}
