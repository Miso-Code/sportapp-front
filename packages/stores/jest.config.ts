export default {
	preset: 'ts-jest',
	setupFilesAfterEnv: ['<rootDir>/config/test/jest.setup.ts'],
	testEnvironment: 'jsdom',
	transform: {
		'^.+\\.tsx?$': ['ts-jest']
	},
	moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
	collectCoverage: true,
	collectCoverageFrom: [
		'<rootDir>/src/**/*.{ts,tsx}',
		'!<rootDir>/src/**/interfaces/**/*.{ts,tsx}',
		'!<rootDir>/src/**/__mocks__/**/*.{ts,tsx}'
	],
	moduleNameMapper: {
		'\\.(gif|ttf|eot|svg|png|jpg)$':
			'<rootDir>/config/test/__mocks__/fileMock.ts',
		'^.+\\.(css|less|scss)$': 'babel-jest',
		'^@/(.*)$': ['<rootDir>/src/$1']
	},
	coverageThreshold: {
		global: {
			branches: 70,
			functions: 70,
			lines: 70,
			statements: 70
		}
	},
	reporters: ['default']
}
