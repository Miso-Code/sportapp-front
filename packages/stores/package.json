{
	"name": "@sportapp/stores",
	"version": "1.0.0",
	"license": "MIT",
	"main": "src/index.ts",
	"type": "module",
	"scripts": {
		"format": "prettier --write \"**/*.{ts,tsx,md,js}\"",
		"lint": "yarn lint:js",
		"lint:js": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
		"pre-commit": "lint-staged",
		"test": "jest --ci --reporters=default --config jest.config.ts --no-cache --coverage --forceExit --detectOpenHandles --logHeapUsage --silent=false --verbose"
	},
	"dependencies": {
		"@react-native-async-storage/async-storage": "^1.23.1",
		"@sportapp/sportapp-repository": "*",
		"@tanstack/react-query": "^5.28.9",
		"@testing-library/jest-dom": "^6.4.2",
		"@testing-library/react": "^14.2.2",
		"@types/jest": "^29.5.12",
		"babel-jest": "^29.7.0",
		"jest": "^29.7.0",
		"prettier": "^3.2.5",
		"react": "^18.2.0",
		"ts-jest": "^29.1.2",
		"ts-jest-mock-import-meta": "^1.2.0",
		"ts-node": "^10.9.2",
		"zustand": "^4.5.2"
	},
	"devDependencies": {
		"@sportapp/eslint-config": "*",
		"@sportapp/prettier-config": "*",
		"@sportapp/typescript-config": "*",
		"eslint": "^8.19.0",
		"lint-staged": "^15.2.2",
		"simple-zustand-devtools": "^1.1.0"
	}
}
