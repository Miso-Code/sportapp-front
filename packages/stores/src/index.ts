export { useAuthStore, initialAuthState } from './auth'
export { useSportSessionStore, initialSportSessionState } from './sportSession'
export { useSportStore, initialSportState } from './sport'
export { useUserStore, initialUserState } from './user'
export {
	usePartnerAuthStore,
	initialAuthState as initialAuthStatePartner
} from './partner/auth'
export {
	useBusinessPartnerStore,
	initialBusinessPartnerState
} from './businessPartner'
export { useAlertStore, initialAlertState } from './alert'
export {
	initialProductPartnerState,
	usePartnerProductStore
} from './partner/products'

export { initialTrainingPlanState, useTrainingPlanStore } from './trainingPlan'

export { initialPaymentPlanState, usePaymentPlanStore } from './paymentPlan'

export { initialSportsmanState, useSportsmanStore } from './sportsman'

export { initialSportEventState, useSportEventStore } from './sportEvents'
