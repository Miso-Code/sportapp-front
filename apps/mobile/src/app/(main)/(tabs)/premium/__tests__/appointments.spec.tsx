import React from 'react'
import renderer, { ReactTestRenderer, act } from 'react-test-renderer'

import Appointments from '../appointments'
import { useSportsmanStore } from '@sportapp/stores'
import { ActivityIndicator } from 'react-native-paper'
import AppointmentCard from '@/components/AppointmentCard'

jest.mock('expo-router')

jest.mock('@/components/AppointmentCard', () => {
	const native = jest.requireActual('react-native')
	return jest.fn((props) => (
		<native.View {...props} testID='appointmentCard' />
	))
})

jest.mock('@sportapp/stores', () => ({
	useSportsmanStore: jest.fn().mockReturnValue({
		trainers: [],
		sportsmanAppointments: [],
		loading: false,
		getAllTrainers: jest.fn(),
		getAllSportsmanAppointments: jest.fn()
	}),
	useAlertStore: jest.fn().mockReturnValue({
		setAlert: jest.fn()
	})
}))

describe('Appointments', () => {
	let component: ReactTestRenderer

	beforeEach(() => {
		component = renderer.create(<Appointments />)
	})

	afterEach(() => {
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should call getAllSportsmanAppointments and getAllTrainers on mount', async () => {
		await act(async () => {
			await Promise.resolve()
		})
		expect(
			useSportsmanStore().getAllSportsmanAppointments
		).toHaveBeenCalled()
		expect(useSportsmanStore().getAllTrainers).toHaveBeenCalled()
	})

	it('should render loading indicator when loading', () => {
		;(useSportsmanStore as unknown as jest.Mock).mockReturnValueOnce({
			...useSportsmanStore(),
			loading: true
		})
		component.update(<Appointments />)
		const activityIndicator = component.root.findByType(ActivityIndicator)
		expect(activityIndicator).toBeTruthy()
	})

	it('should render appointment cards when not loading', async () => {
		const appointments = [
			{
				appointment_id: '1',
				appointment_date: new Date().toISOString(),
				trainer_id: '1',
				appointment_type: 'type',
				appointment_location: 'location'
			},
			{
				appointment_id: '2',
				appointment_date: new Date().toISOString(),
				trainer_id: '2',
				appointment_type: 'type',
				appointment_location: 'location'
			}
		]
		const trainers = [
			{
				trainer_id: '1',
				first_name: 'first',
				last_name: 'last'
			},
			{
				trainer_id: '2',
				first_name: 'first2',
				last_name: 'last2'
			}
		]
		;(useSportsmanStore as unknown as jest.Mock).mockReturnValueOnce({
			...useSportsmanStore(),
			sportsmanAppointments: appointments,
			trainers,
			loading: false
		})

		await act(async () => {
			component.update(<Appointments />)
			await Promise.resolve()
		})

		const appointmentCards = component.root.findAllByType(AppointmentCard)
		expect(appointmentCards).toHaveLength(2)

		for (const [index, card] of appointmentCards.entries()) {
			expect(card.props).toEqual({
				date: appointments[index].appointment_date,
				trainer: `${trainers[index].first_name} ${trainers[index].last_name}`,
				type: appointments[index].appointment_type,
				location: appointments[index].appointment_location
			})
		}
	})
})
