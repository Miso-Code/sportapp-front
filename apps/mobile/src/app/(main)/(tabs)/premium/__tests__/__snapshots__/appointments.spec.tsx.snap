// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 120,
      "paddingHorizontal": 15,
    }
  }
>
  <View>
    <View
      style={
        {
          "gap": 10,
          "marginBottom": 200,
          "width": "100%",
        }
      }
    />
  </View>
</RCTScrollView>
`;
